docs: adapt release documentation according to new pre-release process (#567)
kyma-project/opentelemetry-collector-components: docs/contributor/releasing.md

Reviewer: no future tense

```suggestion
> - If you select `Do-Not-Publish`, the release is not published. However, the action still creates an image and a draft release.
```

Reviewer: no future tense

```suggestion
> - If you select `Force`, it overwrites the existing release and any previously built images. It never overwrites an already published release. For draft releases, the force option recreates the Git tag in the repository, recreates the release notes, and overwrites the previously built image.
```

Reviewer: ```suggestion
Because of the squash-and-merge GitHub workflow, each PR results in a single commit after merging into the main development branch. The PR title becomes the commit message and must adhere to the following template:
```

Reviewer: ```suggestion
- Start with an uppercase, and don't finish with a full stop.
```

Reviewer: ```suggestion
Note that PRs of type `chore` do not appear in the changelog for the release. Therefore, exclude maintenance changes that are not interesting to consumers of the project by marking them as "chore", for example:
```

Reviewer: you call this "changelog" in line 22ff, let's stay consistent 

```suggestion
5. If the previous release was a patch release that contains cherry-picked changes, these changes might appear again in the generated changelog. If there are redundant entries, edit the release description and remove them.
```

This release process covers the steps to release new major and minor versions for the `opentelemetry-collector` with Kyma-specific customizations.

> [!IMPORTANT]

> > Release OCC **before** creating the telemetry-manager pre-release (D-1). The same OCC image version must be used for both the pre-release and the final release of telemetry-manager. Do not release a new OCC version on release day.

Reviewer: ```suggestion
> Release OCC **one day before** creating the telemetry-manager pre-release. You must use the same OCC image version for both the pre-release and the final release of telemetry-manager. Do not release a new OCC version on release day.
```
